.github/workflows/lint.yml

```diff
@@ -0,0 +1,43 @@
+name: Lint
+on:
+  pull_request:
+    paths:
+      - "**.py"
+
+env:
+  PYTHON_VERSION: 3.8
+  LINT_CONFIG: pyproject.toml
+
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+      - name: Setup python $PYTHON_VERSION
+        uses: actions/setup-python@v5
+        with:
+          python-version: $PYTHON_VERSION
+      - name: Get changed files
+        id: changed-files
+        uses: tj-actions/changed-files@v44
+      - name: List changed files
+        run: |
+          echo Changed files: ${{ steps.changed-files.outputs.all_changed_files }}
+      - name: Install python dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install black==24.4.2 pylint==3.1.0 mypy==1.10.0 isort==5.13.2
```

Comment:
I would not recommend installing linters in CI because we need to have the same version installed locally. 
Should we specify development dependencies in the `pyproject.toml` file for Poetry?

Reply:
We can, but we cannot install deps from pyproject.toml using pip. Im not sure if we should use poetry here...

Reply:
Why not? Poetry is our requirement for every microservice